### DISCUSSION TITLE
Various type of input data into forward function in ExactGP

### DISCUSSION THREAD

**Discussion by skiyohara** [cornellius-gp/gpytorch#2603] [Q&A]
Dear gpytorch developer and users,

I have a question regarding the input to the forward function when performing Deep Kernel Learning by inheriting ExactGP.

Upon reading the source code, I noticed that when the forward function has multiple arguments, it concatenates training_inputs and inputs. However, in my case, I want it to receive Data from pytorch_geometric.data (dictionary-style data that maintains the graph structure). Is there an easy way to set it up like this? Additionally, I would like to use lists as forward inputs that should not be concatenated.
Especially, I think I pay careful attention to prediction_strategy.

I also have another question about the "__call__" function in ExactGP. During training, it outputs gpytorch.distributions.MultivariateNormal(mean_x, covar_x) without using train_targets. Why is it able to learn the parameters without any information about train_targets?
I understand that the loss function, ExactMarginalLogLikelihood(model.likelihood, model), receives both the predicted target and the true target for each training batch. However, I am still unclear about the reasoning mentioned above.

Any help will be appreciated.
